cms/templatetags/wagtail_image_lazy_load.py

```diff
@@ -0,0 +1,66 @@
+"""CMS templatetags"""
+
+from urllib.parse import quote_plus
+
+from django import template
+
+from bs4 import BeautifulSoup
+from wagtail.images.templatetags.wagtailimages_tags import image
+from wagtail_lazyimages.templatetags.lazyimages_tags import LazyImageNode
+
+register = template.Library()
+
+
+class CustomLazyImageNode(LazyImageNode):
+    """
+    An extention to Wagtail-lazyimages library's wagtail_lazyimages.templatetags.lazyimages_tags.LazyImageNode
+    class for customizations
+    """
+
+    def render(self, context):
+        """
+        This method overrides LazyImageNode's render method to add file-hashes to image urls for better caching
+
+        Returns:
+            str or None: An image tag with required src (having tiny blurry placeholder image) and
+                data-src (having full-scale image) attribute values
+        """
+        img = self.image_expr.resolve(context)
+        if not img:
+            return ""
+
+        raw_img_tag = super().render(context)
+        parsed_content = BeautifulSoup(raw_img_tag, "lxml").img
```

Comment:
This approach is clean enough in terms of avoiding code repetition, and it's unfortunate that the library doesn't give us extension options, but I'm not sure I agree with this approach of generating the image tag with the library implementation, parsing it as XML, adding attributes, and casting it as a string again. I think I would prefer even a regex replace – after all, we're just tacking on the same `?v=<hash>` to the end of those two attribute values. The XML parsing just seems like more work than we need to be doing.

Reply:
Updated.